orbstation/modules/cargo/cadavers.dm

```diff
@@ -0,0 +1,125 @@
+	name = "Medical Cadaver Crate (Human)"
+	desc = "A preserved medical cadaver, for dissections, spare organs, or experiments. This one is a human."
+	contains = list(/mob/living/carbon/human)
+
+/datum/supply_pack/medical/cadaver/jelly
+	name = "Medical Cadaver Crate (Jellyperson)"
+	desc = "A preserved medical cadaver, for dissections, spare organs, or experiments. This one is a jellyperson."
+	contains = list(/mob/living/carbon/human/species/jelly)
+
+/datum/supply_pack/medical/cadaver/moth
+	name = "Medical Cadaver Crate (Moth)"
+	desc = "A preserved medical cadaver, for dissections, spare organs, or experiments. This one is a moth."
+	contains = list(/mob/living/carbon/human/species/moth)
+
+/datum/supply_pack/medical/cadaver/pod
+	name = "Medical Cadaver Crate (Podperson)"
+	desc = "A preserved medical cadaver, for dissections, spare organs, or experiments. This one is a podperson."
+	contains = list(/mob/living/carbon/human/species/pod)
+
+/datum/supply_pack/medical/cadaver/rat
+	name = "Medical Cadaver Crate (Ratfolk)"
+	desc = "A preserved medical cadaver, for dissections, spare organs, or experiments. This one is a ratfolk."
+	contains = list(/mob/living/carbon/human/species/ratfolk)
+
+/datum/supply_pack/medical/cadaver/tiziran
+	name = "Medical Cadaver Crate (Tiziran)"
+	desc = "A preserved medical cadaver, for dissections, spare organs, or experiments. This one is a tiziran."
+	contains = list(/mob/living/carbon/human/species/lizard)
+
+// I'm making this a subtype of coffin solely so vampires can sleep in them.
+/obj/structure/closet/crate/coffin/stasis
+	name = "stasis coffin"
+	desc = "An advanced coffin used to keep corpses fresh for extended transport. Once opened, it will permanently stop functioning."
+	icon = 'orbstation/icons/obj/coffin.dmi'
+	icon_state = "stasiscoffin"
+	resistance_flags = NONE
+	max_integrity = 70
+	material_drop = /obj/item/stack/sheet/iron
+	material_drop_amount = 5
+	open_sound = 'sound/machines/closet_open.ogg'
+	close_sound = 'sound/machines/closet_close.ogg'
+	open_sound_volume = 25
+	close_sound_volume = 50
+	can_install_electronics = FALSE
+	mob_storage_capacity = 1 //no mass stasis, sorry
+	var/functional = TRUE //whether the coffin still works or not
+
+/obj/structure/closet/crate/coffin/stasis/closet_update_overlays(list/new_overlays)
+	. = new_overlays
+	if(!opened && functional)
+		. += "stasisactive"
+	if(manifest)
+		. += "manifest"
+
+/obj/structure/closet/crate/coffin/stasis/open(mob/living/user, force = FALSE)
+	if(!functional)
+		return ..()
+	functional = FALSE
+	for(var/mob/living/carbon/corpse in src) //unfreeze the organs of any contained carbon
+		for(var/obj/item/organ/part as anything in corpse.internal_organs)
+			part.organ_flags &= ~ORGAN_FROZEN
+	return ..()
+
+/obj/structure/closet/crate/coffin/stasis/close(mob/living/user)
+	. = ..()
+	if(!functional)
+		return
+	for(var/mob/living/carbon/corpse in src) //freeze the organs of any contained carbon
```

Comment:
Is this ever called or required?
We set `functional` to `FALSE` when it is opened and do the stasis application manually when anything is placed into the crate.
When is it possible to close a still-functioning stasis coffin?

Reply:
If this never actually executes we might not need the `functional` variable at all.

Reply:
Oops. This was an artifact from an older version of this code where this made more sense. Removed.